pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -355,16 +283,18 @@ func (am *Alertmanager) ApplyConfig(dbCfg *ngmodels.AlertConfiguration) error {
 		return fmt.Errorf("failed to parse Alertmanager config: %w", err)
 	}
 
-	am.reloadConfigMtx.Lock()
```

Comment:
Do not conflate `ApplyConfig` with `applyConfig` - this one also needs a lock because all public functions should be safe to be called concurrently.